src/combine_flowfiles.py

```
@@ -0,0 +1,207 @@
+        with open(output_path, "r") as f:
+            combined_content = f.read()
+            print(combined_content)
+
+        # Show deduplication summary
+        combined_df = pd.read_csv(output_path)
+        overlap_ids = ["id002", "id005", "id008"]
+        print("\n=== Deduplication Summary ===")
+        for feature_id in overlap_ids:
+            values = []
+            for i, df in enumerate(test_data, 1):
+                if feature_id in df["featureid"].values:
+                    val = df[df["featureid"] == feature_id]["discharge"].values[0]
+                    values.append(f"File {i}: {val}")
+            final_val = combined_df[combined_df["featureid"] == feature_id]["discharge"].values[0]
+            print(f"{feature_id}: {' vs '.join(values)} → Kept: {final_val} (max)")
+
+        print("\nTest completed successfully!")
+
+    finally:
+        shutil.rmtree(temp_dir)
+
+
+def combine_flowfiles(flowfile_paths, output_path):
+    """
+    Combine multiple flowfiles into a single CSV file.
+
+    Args:
+        flowfile_paths: List of S3 or local paths to flowfiles
+        output_path: Path where combined flowfile will be saved
+    """
+    all_dfs = []
+    header = None
+
+    for flowfile_path in flowfile_paths:
+        try:
+            print(f"  Reading: {flowfile_path}")
+            with fsspec.open(flowfile_path, "r") as f:
+                df = pd.read_csv(f)
+                if header is None:
+                    header = df.columns.tolist()
+                all_dfs.append(df)
+        except Exception as e:
+            print(f"  Warning: Could not read {flowfile_path}: {e}")
+            continue
+
+    if not all_dfs:
+        raise ValueError("No flowfiles could be read")
+
+    combined_df = pd.concat(all_dfs, ignore_index=True)
+
+    # Deduplicate rows based on the first column, keeping the maximum value in the second column
+    if len(combined_df.columns) >= 2:
```

Can you explain why we are only doing this when more than 2 columns?

---

Well we need at least two columns and I did greater than or equal too in case there were any flowfiles with additional columns. Ideally we could just refer to the columns by name explicitly but I have run into flowfiles that have non-standard column names so I'm not sure what the solution is.

---

What about when there are less than 2 columns, in that case we should raise the error, right? or else we just don't need the if statement at all?

---

Yes then we should raise an error will add